scripts/prepare_vehicle_electricity_demand.py

```diff
@@ -0,0 +1,115 @@
+# coding: utf-8
+r"""
+Inputs
+-------
+input_dir : str
+    ``raw/time_series/vehicle_charging``: Path to directory where csv files containing electric
+    vehicle charging demand time series are placed
+output_file : str
+    ``results/_resources/ts_load_electricity_vehicles.csv``: Path incl. file name to prepared time
+    series
+
+Outputs
+---------
+pandas.DataFrame
+    Normalized electric vehicle charging demand for regions "B" and "BB" for all years provided in
+    `input_dir`.
+
+Description
+-------------
+This script prepares electric vehicle charging demand time series for the regions Berlin
+and Brandenburg. The time series have been created before with simBEV
+(https://github.com/rl-institut/simbev). The charging strategy is "greedy", i.e. batteries are
+charged with maximum power until they are fully charged or removed.
+
+"""
+
+import sys
+import pandas as pd
+import os
+import oemof_b3.tools.data_processing as dp
+
+# global variables
+TS_VAR_UNIT = "None"
+TS_SOURCE = "created with simBEV"
+TS_COMMENT = "https://github.com/rl-institut/simbev"
+
+
+def prepare_vehicle_time_series(input_dir):
+    r"""
+    Prepares and formats electric vehicle charging demand time series for regions 'B' and 'BB'.
+
+    Parameters
+    ----------
+    input_dir : str
+        Path to directory where csv files containing electric vehicle charging demand time series
+        are placed
+
+    Returns
+    -------
+    ts_prepared : pd.DataFrame
+        Contains electric vehicle charging demand time series in the format of time series template
+        of oemof-B3
+
+    """
+    # initialize data frame
+    df = pd.DataFrame()
+
+    # loop through files in `input_dir`
+    for filename in os.listdir(input_dir):
+        # get year and region from file name
+        split_filename = os.path.splitext(os.path.basename(filename))[0].split("_")
+        region, year = split_filename[2], int(split_filename[3])
+        path = os.path.join(input_dir, filename)
+
+        # read data from file, copy and superfluous drop last time step
+        ts_raw = pd.read_csv(
+            path, index_col=1, sep=";", decimal=",", parse_dates=True
+        ).drop(columns=["Unnamed: 0"], axis=1)
+        ts = ts_raw.copy()
```

Comment:
As you do not use ts_raw later, this copy seems unnecessary. Correct me if I am wrong.

Reply:
Ah yes, I never know, when the copy warning is coming.. but you are right, it's not necessary.